sqlobject/src/test/java/org/jdbi/v3/sqlobject/customizer/TestAllowUnusedBindings.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject.customizer;
+
+import org.jdbi.v3.core.rule.H2DatabaseRule;
+import org.jdbi.v3.core.statement.UnableToCreateStatementException;
+import org.jdbi.v3.sqlobject.SqlObjectPlugin;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestAllowUnusedBindings {
+    @Rule
+    public H2DatabaseRule dbRule = new H2DatabaseRule().withPlugin(new SqlObjectPlugin());
+    private UnusedBindingDao dao;
+
+    @Before
+    public void setUp() {
+        dao = dbRule.openHandle().attach(UnusedBindingDao.class);
+    }
+
+    @Test
+    public void testAllowed() {
+        assertThat(dao.allowed("42")).isTrue();
+    }
+
+    @Test
+    public void testDisallowed() {
+        assertThatThrownBy(() -> dao.disallowed("43"))
```

Comment:
I'd set it to `true` explicitly on the handle instance to really prove that the annotation overrides it

Reply:
Setting it on the handle _after_ the dao instance has been created is a moot action, isn't it?

Reply:
made it `onDemand` instead